src/main/kotlin/uk/gov/gdx/datashare/repositories/AcquirerEventRepository.kt

```diff
@@ -62,4 +62,12 @@ interface AcquirerEventRepository : CrudRepository<AcquirerEvent, UUID> {
   fun softDeleteById(id: UUID, deletionTime: LocalDateTime)
 
   fun countByDeletedAtIsNull(): Int
+
+  @Query(
+    "SELECT ae.supplier_event_id  " +
+      "FROM acquirer_event ae JOIN supplier_event se on ae.supplier_event_id = se.id and se.event_consumed = false " +
+      "GROUP BY ae.supplier_event_id " +
+      "HAVING SUM( CASE when ae.deleted_at is null then 0 else 1 END ) = COUNT(*)",
+  )
+  fun findAllByDeletedEventsForAllAcquirers(): List<UUID>
```

Comment:
As this is returning supplier event IDs, should it be in the supplier event repo?

Also for the sum logic, could we have 1 when deleted at is null, and then instead we check that the value is 0 not count(*)?

Reply:
yes